src/accumulator.jl

```diff
@@ -9,30 +9,22 @@ end
 Accumulator(::Type{T}, ::Type{V}) where {T,V<:Number} = Accumulator{T,V}(Dict{T,V}())
 counter(T::Type) = Accumulator(T,Int)
 
-counter(dct::Dict{T,Int}) where {T} = Accumulator{T,Int}(copy(dct))
+counter(dct::Dict{T,V}) where {T,V<:Integer} = Accumulator{T,V}(copy(dct))
 
 """
-    counter{T}(seq::AbstractArray)
+    counter(seq)
 
 Returns an `Accumulator` object containing the elements from `seq`.
 """
-function counter(seq::AbstractArray{T}) where T
-    ct = counter(T)
+function counter(seq)
+    ct = counter(eltype(seq))
     for x in seq
-        push!(ct, x)
+        inc!(ct, x)
     end
     return ct
 end
 
-function counter(gen::T) where {T<:Base.Generator}
-    ct = counter(Base._default_eltype(T))
```

Comment:
This code was just added recently (by @cstjean).  I'm going to suggest leaving it in until the `eltype` of a generator can be more easily determined.  @cstjean, any thoughts?

Reply:
Is there a reason to take it out? It provides much better typing of the accumulator.

Reply:
oop, I actually  though I had it covered by adding general iterators.
I didn't know `_default_eltype` was different to `eltype`, but now I see it is different.

With that said:
I wonder why Base doesn't have `eltype(T::Type{Generator})=_default_eltype(T)`.
The fact that Base does not do this leaves me worried.
It is an undocumented, unexported method,
that seems like it should be used and yet, it is not.

I will put that back in.
This is closely related to  #248.